src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -328,12 +328,31 @@ typedef struct server_vnodes_t {
 } server_vnodes_t;
 static	server_vnodes_t server_vnodes;
 
+/**
+ * @brief
+ * 	This is a function that logs the contents of the list
+ *	headed by 'phead'.
+ *
+ * @param[in] head_str - some string that gets printed spearheading the list.
+ * @param[in] phead - header of the list to be printed.
+ *
+ * @return	void
+ *
+ */
 void
 print_svrattrl_list(char *head_str, pbs_list_head *phead)
 {
 	svrattrl *plist = NULL;
 	int i;
 
+	if ((head_str == NULL) || (phead == NULL) || (log_event_mask == NULL)) {
+		return;
+	}
+
+	if ((*log_event_mask & PBSEVENT_DEBUG3) == 0) {
```

Comment:
Bhroam, since your opposition is due to the fact that this breaks logging library encapsulation, how about if I just introduce a new logging library function that will check if a given event type gets logged by pbs. Then in the PBS python code, I can just call this log library function. I'm thinking of the following:

Libpython/pbs_python_svr_internal.c:

!       if ((*log_event_mask & PBSEVENT_DEBUG3) == 0) {
                return;
-       }

        plist = (svrattrl *)GET_NEXT(*phead);
        i = 0;
--- 345,356 ----
        svrattrl *plist = NULL;
        int i;

! 
!       if (!log_event_type_recorded(PBSEVENT_DEBUG3))
                return;

---
include/log.h:
+ extern int log_event_type_recorded(int type);
---
Liblog/log_event.c

NOTE: And this new function can actually replace the reference to 'log_event_mask' under log_event():

  log_event(int eventtype, int objclass, int sev, const char *objname, const char *text)
  {
!       if (((eventtype & PBSEVENT_FORCE) == 0) &&
!               ((*log_event_mask & eventtype) == 0))
!               return;         /* not logging this type of event */
!       else
                log_record(eventtype, objclass, sev, objname, text);
  }
--- 118,123 ----
  void
  log_event(int eventtype, int objclass, int sev, const char *objname, const char *text)
  {
!       if (log_event_type_recorded(eventtype))
                log_record(eventtype, objclass, sev, objname, text);
  }
---
+ int
+ log_event_type_recorded(int eventtype)
+ {
+       if (((eventtype & PBSEVENT_FORCE) != 0) ||
+           ((*log_event_mask & eventtype) != 0))
+               return (1);
+
+       return (0);
+ }